metricflow/sql_clients/big_query.py

```diff
@@ -51,19 +51,34 @@ def from_connection_details(url: str, password: Optional[str]) -> BigQuerySqlCli
 
         return BigQuerySqlClient(password=password)
 
-    def __init__(self, password: str, query_string: str = "") -> None:
+    def __init__(self, project_id: str = "", password: Optional[str] = None, query_string: str = "") -> None:
         """Creates a new BigQueryDBClient and tags it for tracing as big query."""
         # Without pool_pre_ping, it's possible for timed-out connections to be returned to the client and cause errors.
         # However, this can cause increase latency for slow engines.
         self._password = password
-        super().__init__(engine=BigQuerySqlClient._create_bq_engine(query_string=query_string, password=password))
+        super().__init__(
+            engine=BigQuerySqlClient._create_bq_engine(
+                query_string=query_string, project_id=project_id, password=password
+            )
+        )
 
     @staticmethod
-    def _create_bq_engine(query_string: str, password: str) -> sqlalchemy.engine.Engine:
-        """Create the connection engine in SqlAlchemy to connect to BQ"""
+    def _create_bq_engine(
+        query_string: str, project_id: str = "", password: Optional[str] = None
+    ) -> sqlalchemy.engine.Engine:
+        """Create the connection engine in SqlAlchemy to connect to BQ
+
+        There are 2 methods of creating the BigQuery Engine,
+        1. Using a service account JSON credential which will load into the credentials_info (Recommended for production use).
+        2. Using ADC (Application Default Credentials) which allows a user to run `gcloud auth application-default login` to auth.
+           This method would pass None to credentials_info and it will aggregate the credentials via,
+           - Google's decision tree (https://google.aip.dev/auth/4110)
+        """
         return sqlalchemy.create_engine(
-            f"{SqlDialect.BIGQUERY.value}://" + "/?" + query_string,
-            credentials_info=json.loads(password),  # "password := string representation of keyfile.json",
+            f"{SqlDialect.BIGQUERY.value}://{project_id}" + "/?" + query_string,
+            credentials_info=json.loads(password)
```

Comment:
small nit: these one line if statements are easier to understand when they're written on one line. I understand the auto formatter formatted it this way. If we put the password definition comment in the docstring instead of here, this all could go on one line.